### PR TITLE
Added new image segmentation lab exercise (Car segmentation)

### PR DESCRIPTION
**Pull request by kenghooi-teoh**
When you added a new program / programs , put a cross on the  OS that you have test it on  
(Example: [ X ] Windows  )

[X] Windows  
[  ] Linux Ubuntu  
[  ] Mac  
[  ] Others  (State here -> xxx )  

Assign a reviewer to review your code.

### CODE REVIEW

File: dl4j-cv-labs/src/main/java/global/skymind/solution/classification/FoodClassification.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        // UI server setup
+        UIServer uiServer = UIServer.getInstance();
+        StatsStorage statsStorage = new InMemoryStatsStorage();
+        uiServer.attach(statsStorage);
+
+        /*
+        * LOAD MODEL
+        * */
+        log.info("\n\nLoading org.deeplearning4j.transferlearning.resnet50...\n\n");
+//        ZooModel zooModel = VGG16.builder().build();
```
Comment (peirkern-skymind) [inline dl4j-cv-labs/src/main/java/global/skymind/solution/classification/FoodClassification.java]: cleanup

File: dl4j-cv-labs/src/main/java/global/skymind/solution/classification/FoodClassification.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                                .nIn(2048).nOut(numClasses)
+                                .weightInit(new NormalDistribution(0,0.2*(2.0/(4096+numClasses))))
+//                                .weightInit(WeightInit.XAVIER)
+                                .activation(Activation.SOFTMAX).build(),
+                        "flatten_1")
+                .build();
+        log.info(resnet50Transfer.summary());
+
+        //Dataset iterators
+//        FoodDataSetIterator.setup(batchSize);
```
Comment (peirkern-skymind) [inline dl4j-cv-labs/src/main/java/global/skymind/solution/classification/FoodClassification.java]: cleanup

File: dl4j-cv-labs/src/main/java/global/skymind/solution/classification/FoodClassification.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        //Construct a new model with the intended architecture and print summary
+        ComputationGraph resnet50Transfer = new TransferLearning.GraphBuilder(resnet50)
+                .fineTuneConfiguration(fineTuneConf)
+                .setFeatureExtractor(featureExtractionLayer) //the specified layer and below are "frozen"
+                .removeVertexKeepConnections("fc1000") //replace the functionality of the final vertex
+                .addLayer("fc1000",
+                        new OutputLayer.Builder(LossFunctions.LossFunction.MCXENT)
+                                .nIn(2048).nOut(numClasses)
+                                .weightInit(new NormalDistribution(0,0.2*(2.0/(4096+numClasses))))
+//                                .weightInit(WeightInit.XAVIER)
```
Comment (peirkern-skymind) [inline dl4j-cv-labs/src/main/java/global/skymind/solution/classification/FoodClassification.java]: cleanup

File: dl4j-cv-labs/src/main/java/global/skymind/solution/classification/FoodDataSetIterator.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                new Pair<>(flipTransform2, 0.8),
+                new Pair<>(rotateTransform, 0.5));
+
+        ImageTransform transform = new PipelineImageTransform(pipeline, shuffle);
+
+        ImageRecordReader recordReader = new ImageRecordReader(height,width,channels,labelMaker);
+        recordReader.initialize(split, transform);
+
+        DataSetIterator iter = new RecordReaderDataSetIterator(recordReader, batchSize, 1, numClasses);
+//        iter.setPreProcessor( new VGG16ImagePreProcessor());
```
Comment (peirkern-skymind) [inline dl4j-cv-labs/src/main/java/global/skymind/solution/classification/FoodDataSetIterator.java]: cleanup

File: dl4j-cv-labs/src/main/java/global/skymind/solution/segmentation/car/CustomLabelGenerator.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import org.datavec.image.loader.NativeImageLoader;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+public class CustomLabelGenerator implements PathLabelGenerator {
```
Comment (peirkern-skymind) [inline dl4j-cv-labs/src/main/java/global/skymind/solution/segmentation/car/CustomLabelGenerator.java]: refactor this class, there're quite a few of CustomLabelGenerator, kinda redundant.

**Comment by peirkern-skymind** [comment]
pending: issue yet to resolve, crash half way.

File: dl4j-cv-labs/src/main/java/global/skymind/solution/segmentation/car/CustomLabelGenerator.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
+//package global.skymind.solution.segmentation.car;
```
Comment (peirkern-skymind) [inline dl4j-cv-labs/src/main/java/global/skymind/solution/segmentation/car/CustomLabelGenerator.java]: legacy codes?

File: dl4j-cv-labs/src/main/java/global/skymind/solution/segmentation/car/PretrainedUNET.java [hunk 7]
```diff
@@ -108,120 +106,113 @@ public static void main(String[] args) throws IOException {
...
-                frame,
-                1,
-                height,
-                width,
-                1
-        );
-
         // EXPORT IMAGES
-        File exportDir = new File("export");
+//        File exportDir = new File("export");
```
Comment (peirkern-skymind) [inline dl4j-cv-labs/src/main/java/global/skymind/solution/segmentation/car/PretrainedUNET.java]: legacy codes?

File: dl4j-cv-labs/src/main/java/global/skymind/solution/segmentation/cell/CustomLabelGenerator.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
+//package global.skymind.solution.segmentation.cell;
```
Comment (peirkern-skymind) [inline dl4j-cv-labs/src/main/java/global/skymind/solution/segmentation/cell/CustomLabelGenerator.java]: legacy codes?

**Comment by peirkern-skymind** [comment]
FYA @kenghooi-teoh

**Comment by choowilson** [comment]
missing carsegmentation url at config.properties.. dataset not downloaded

**Pull request merged**